### DISCUSSION TITLE
Accepting input from both microphone and another application

### DISCUSSION THREAD

**Discussion by cherryblossom000** [ExistentialAudio/BlackHole#358] [Q&A]
I’m trying to send sound output from one application (e.g. QuickTime Player), as well as input from my microphone, to Discord.

My mac is using a multi-output device, called ‘HP + BH (O)’, with my headphones and BlackHole 2ch for sound output so I can both hear the sound from QuickTime Player and send it to Discord. This works fine.

However, I can’t seem to get multiple inputs (from both my microphone and BlackHole) to work. I created an aggregate device (called ‘HP + BH (IO)’) with my microphone, headphones, and BlackHole:

![screenshot of aggregate device configuration](https://user-images.githubusercontent.com/31467609/114253497-c8cb5600-99ed-11eb-83c7-3639f519e24f.png)

I tried the following for Discord input devices:

- External Microphone: Discord receives input from only my microphone (expected)
- BlackHole 2ch: Discord receives input from only the other app (expected)
- HP + BH (IO): Discord receives input from only my microphone (unexpected, I expected it to receive input from the other app as well)

I did this with both the ‘HP + BH (O)’ and BlackHole 2ch device for my Mac‘s output, which yielded the same results (except that with BlackHole 2ch, I wasn’t able to hear the output from the other app, which was expected).

Am I doing this right? Is there a way to receive input from multiple sources?

**Comment by thibosLN** [comment]
I have a simpler version of this same problem. I want to present a Quicktime movie in Zoom using screen sharing.  Only the video comes through, not the audio.  I have selected Black Hole as the input source (to replace the default built-in microphone) in Zoom, but that only mutes the microphone without supplying the soundtrack from the QT movie.  This is on a PowerBookPro with M1 silicon running Big Sur 11.2.  Privacy System Preference gives microphone access to Zoom, but Black Hole is not in the list of apps for which preferences may be set.   Any suggestions?

**Comment by tonywatkins** [comment]
I don't know whether this will help your particular issue, but I've finally cracked this issue for recording external mic and system audio into Adobe Audition: see my post here: https://github.com/ExistentialAudio/BlackHole/discussions/395
